### PR TITLE
Scala 2.13

### LINKED ISSUE #82: Release for Scala 2.13.x

### PR DESCRIPTION
**Pull request by ghik**
Fixes #82 

Also:
* Dropped support for Scala 2.11
* Changed versioning scheme to semantic versioning - next version released from `master` will be 2.0.0 while `1.34.x` will continue as `1.x` and the next version from that branch will be 1.35.0

Two modules (`commons-akka` and `commons-mongo`) are currently blocked by dependencies and therefore are excluded from the 2.13 build.

### CODE REVIEW

File: commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala [hunk 1]
```diff
@@ -253,7 +255,7 @@ object SharedExtensions extends SharedExtensions {
       fut.transform(s, f)(RunNowEC)
 
     def transformNow[S](f: Try[A] => Try[S]): Future[S] =
-      fut.transformTry(f)(RunNowEC)
+      fut.transform(f)(RunNowEC)
```
Comment (ddworak) [inline commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala]: Any thoughts on `RunNowEC` vs `scala.concurrent.ExecutionContext.parasitic` in 2.13?

File: commons-core/jvm/src/main/scala/com/avsystem/commons/jiop/ScalaJDoubleStream.scala [hunk 2]
```diff
@@ -117,8 +116,8 @@ final class ScalaJDoubleStream(private val jStream: JDoubleStream) extends AnyVa
   def toArray: Array[Double] =
     jStream.toArray
 
-  def to[Col[_]](implicit cbf: CanBuildFrom[Nothing, Double, Col[Double]]): Col[Double] = {
-    val b = cbf.apply()
+  def to[C](implicit fac: Factory[Double, C]): C = {
```
[the remark this answers is not in the document]
Comment (ghik) [reply commons-core/jvm/src/main/scala/com/avsystem/commons/jiop/ScalaJDoubleStream.scala]: That would box my value classes.

File: commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala [hunk 3]
```diff
@@ -40,13 +41,13 @@ trait SharedExtensions extends CompatSharedExtensions {
...
-    new TraversableOnceOps(coll)
+  implicit def iterableOnceOps[C[X] <: IterableOnce[X], A](coll: C[A]): IterableOnceOps[C, A] =
+    new IterableOnceOps(coll)
 
-  implicit def traversableOps[C[X] <: BTraversable[X], A](coll: C[A]): TraversableOps[C, A] = new TraversableOps(coll)
+  implicit def traversableOps[C[X] <: BIterable[X], A](coll: C[A]): IterableOps[C, A] = new IterableOps(coll)
 
-  implicit def pairTraversableOnceOps[C[X] <: TraversableOnce[X], K, V](coll: C[(K, V)]): PairTraversableOnceOps[C, K, V] =
-    new PairTraversableOnceOps(coll)
+  implicit def pairIterableOnceOps[C[X] <: IterableOnce[X], K, V](coll: C[(K, V)]): PairIterableOnceOps[C, K, V] =
```
[the remark this answers is not in the document]
Comment (ghik) [reply commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala]: Still useful in 2.12

File: commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala [hunk 4]
```diff
@@ -56,8 +57,9 @@ trait SharedExtensions extends CompatSharedExtensions {
 
   implicit def orderingOps[A](ordering: Ordering[A]): OrderingOps[A] = new OrderingOps(ordering)
 }
+object SharedExtensions extends SharedExtensions
 
-object SharedExtensions extends SharedExtensions {
+object SharedExtensionsUtils extends SharedExtensions {
```
[the remark this answers is not in the document]
Comment (ghik) [reply commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala]: Scala 2.13 doesn't like the fact that `object SharedExtensions extends SharedExtensions` **and** `trait SharedExtensions` does an `import SharedExtensions._` at the same time.
Not sure why.

File: commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala (hunk 1 above)
Comment (ghik) [reply commons-core/src/main/scala/com/avsystem/commons/SharedExtensions.scala]: We can think about it when we drop 2.12.

**Pull request merged**